lib/model/bloc/transactions/ProfitLossListProvider.dart

```diff
@@ -0,0 +1,115 @@
+import 'dart:async';
+
+import 'package:farmsmart_flutter/model/bloc/transactions/TransactionToProfitLossListItemViewModel.dart';
+import 'package:farmsmart_flutter/model/model/TransactionEntity.dart';
+import 'package:farmsmart_flutter/model/model/crop_entity.dart';
+import 'package:farmsmart_flutter/model/model/loading_status.dart';
+import 'package:farmsmart_flutter/model/repositories/crop/CropRepositoryInterface.dart';
+import 'package:farmsmart_flutter/model/repositories/transaction/TransactionRepositoryInterface.dart';
+import 'package:farmsmart_flutter/ui/profitloss/ProfitLossList.dart';
+import 'package:farmsmart_flutter/ui/profitloss/ProfitLossListItem.dart';
+import 'package:intl/intl.dart';
+import '../ViewModelProvider.dart';
+import 'TransactionToRecordTransactionViewModel.dart';
+
+
+class _Strings {
+  static const generalItemTag = "General";
+  static const currentcyTag = "GBP";
+}
+
+class ProfitLossListProvider
+    implements ViewModelProvider<ProfitLossListViewModel> {
+  final TransactionRepositoryInterface _transactionsRepository;
+  final CropRepositoryInterface _cropRepository;
+  ProfitLossListViewModel _snapshot;
+  final StreamController<ProfitLossListViewModel> _controller =
+      StreamController<ProfitLossListViewModel>.broadcast();
+  List<String> _taglist = [];
+  String _title = "";
+
+  ProfitLossListProvider({
+    TransactionRepositoryInterface transactionsRepository,
+    CropRepositoryInterface cropRepository,
+  }) : this._transactionsRepository = transactionsRepository, this._cropRepository = cropRepository;
+
+  @override
+  StreamController<ProfitLossListViewModel> observe() {
+    return _controller;
+  }
+
+  @override
+  ProfitLossListViewModel snapshot() {
+    return _snapshot;
+  }
+
+  @override
+  ProfitLossListViewModel initial() {
+    if (_snapshot == null) {
+      _taglist= _getTagList([]);
+      _transactionsRepository.observeProfile(null).listen((transactions) {
```

Comment:
for now I add null here, I need to integrate with the profile repo when that's done.